doc/src/guide/protocols.asciidoc

```diff
@@ -85,15 +94,21 @@ the normal `gen_statem` execution loop.
 %% Exports of other gen_statem callbacks here.
 
 start_link(Ref, Transport, Opts) ->
-	{ok, proc_lib:spawn_link(?MODULE, init, [{Ref, Transport, Opts}])}.
+	{ok, proc_lib:start_link(?MODULE, init, [{Ref, Transport, Opts}])}.
 
-init({Ref, Transport, _Opts = []}) ->
+init({Ref, Transport, _Opts}) ->
 	%% Perform any required state initialization here.
+	ok = proc_lib:init_ack(self()),
```

Comment:
I don't think this example should use start_link/init_ack. There's no real value in having a connection process make the connection supervisor wait for it to be setup. It can be mentioned above, but the need for start_link should be very low.

Reply:
Ok, no problem.

Reply:
See last commit.